src/ht.c

```diff
@@ -15,11 +17,30 @@ const int ht_primes_sizes[] = {
 	389357, 467237, 560689, 672827, 807403, 968897, 1162687,
 	1395263, 1674319, 2009191, 2411033, 2893249, 3471899,
 	4166287, 4999559, 5999471, 7199369
-#else
-	1024,
-#endif
 };
 
+#define HASHFN(ht, k) ((ht)->hashfn ? (ht)->hashfn (k) : (ut32)(ut64)(k))
```

Comment:
i dont like having so many macros

Reply:
Why? Do you prefer static functions? Or dup code? I did macros to be sure the code was inlined and because they are very small

Reply:
the reason for those macros are mainly to avoid nullderefs, what about making it into a function that wraps the call instead? like this: 

```
static inline void *wrap(void *a, void *b) {
  return a? a(b): NULL;
}
```

and do `wrap(ht->dupkey, k)` instead of having to dereference the pointer twice. obvously a compiler can do good optimization in both use cases, but looks cleaner to me than messing all those pointer calls with macros

Reply:
I can use `static inline` functions, but I can't wrap all of them in just one `wrap` function, because some of them by default return 0, other by default just return the argument, etc.